### DISCUSSION TITLE
Extending Discounts to support more use cases

### DISCUSSION THREAD

**Discussion by titusdmoore** [medusajs/medusa#4869] [Feature Requests]
### Overview
As it currently stands Discounts are in a good spot with several ways to apply discounts. That being said, to get in line with other ecommerce solutions, there are some additional features that can be added. 

Additional Discounts:
- Product Variation discounts
- Quantity/Bulk Pricing Discounts
- Other general cart state discounts (e.g. buy one, get one)

### Challenges
I am sure that there will be challenges that I cannot foresee, but the main one that is worth mentioning here is that there currently not a discount that uses cart state, so design decisions will need to be made as to how best to accomplish this. 

### Drawbacks
The main drawback I have identified is depending on how we approach handling product variations discounts, we may introduce a breaking change with existing product examples.

**Comment by olivermrbl** [comment]
@titusdmoore, thanks for submitting this feature request :)

A more extensible Discounts API is already on our radar. We expect to rethink the domain's functionality as we modularize it into a standalone package - similar to how we've done with [Products](https://github.com/medusajs/medusa/tree/develop/packages/product). Your input is much appreciated!

Unfortunately, I can't give you a timeline right now. I hope I can share some more details in a couple of months. And if something changes in the meantime, I'll make sure to revert back to you.

**Reply by piereligio** [reply]
Any news? I'd really like being able to offer bundle discounts. Maybe I'll just work it around by making ad hoc bundle products for now.

**Reply by Saba-Shavidze** [reply]
@piereligio any updates?

**Reply by piereligio** [reply]
> @piereligio any updates?

I simply listed products which were bundles, so that I could have a discount price on those

**Comment by titusdmoore** [comment]
For anyone stumbling across this discussion, it appears that the new promotion module that seems to be shipping with version 2.0 will have some of the features discussed here. 
Source: https://medusajs.com/promotion-module/

**Reply by Snorlax-Chan** [reply]
(However, it cannot be tested yet, so it is uncertain whether it supports bundle sales, multiple discount codes, and similar scenarios.)

**Comment by alessioacella** [comment]
will this be compatible with Medusa v1.x version?
